# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: security/certverifier/NSSCertDBTrustDomain.h
# Commit: dc509eab3ed7
# Full Hash: dc509eab3ed7a513258723eb46b25f33a610e720
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2025-07-11 16:36:21
# Regressor Bug: 1953221
# File Overlap Count: 3
# Description:
#   Bug 1953221 - don't accept SCTs with a timestamp after the distrustAfter date for the root r=jschanck
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D256720
# ==============================================================================

diff -r 5b86ae2c81f9 -r dc509eab3ed7 security/certverifier/NSSCertDBTrustDomain.h
--- a/security/certverifier/NSSCertDBTrustDomain.h	Thu Jul 10 18:37:49 2025 +0000
+++ b/security/certverifier/NSSCertDBTrustDomain.h	Thu Jul 10 19:04:10 2025 +0000
@@ -249,6 +249,9 @@
 
   OCSPFetchStatus GetOCSPFetchStatus() { return mOCSPFetchStatus; }
   IssuerSources GetIssuerSources() { return mIssuerSources; }
+  Maybe<mozilla::pkix::Time> GetDistrustAfterTime() {
+    return mDistrustAfterTime;
+  }
 
  private:
   Result CheckCRLite(
@@ -329,6 +332,7 @@
 
   OCSPFetchStatus mOCSPFetchStatus;
   IssuerSources mIssuerSources;
+  Maybe<mozilla::pkix::Time> mDistrustAfterTime;
 };
 
 }  // namespace psm